starcluster/balancers/sge/__init__.py

```diff
@@ -749,22 +747,31 @@ def _eval_add_node(self):
                 assert count > 0, \
                     "There should be 1 or more jobs waiting for too long."
                 slots_jobs_ratio = float(total_slots) / len(running_jobs)
-                avg_job_duration_h = min(1,
-                                         self.stat.avg_job_duration() / 3600.0)
+                avg_job_duration = self.stat.avg_job_duration() or 3600
+                avg_job_duration_h = min(1, avg_job_duration / 3600.0)
                 required_slots = count * avg_job_duration_h * slots_jobs_ratio
                 need_to_add = int(math.ceil(required_slots / slots_per_host))
+                assert need_to_add > 0
             else:
                 log.info("Existing nodes are not running jobs, hence "
                          "StarCluster cannot make a reliable estimate.")
                 need_to_add = 1
 
-        max_add = self.max_nodes - len(self._cluster.running_nodes)
-        need_to_add = min(self.add_nodes_per_iteration, need_to_add, max_add)
-        if need_to_add < 1:
-            return False
+        max_add = self.max_nodes - num_nodes
+        return min(self.add_nodes_per_iteration, need_to_add, max_add)
 
+    def _eval_add_node(self):
+        """
+        This function inspects the current state of the SGE queue and decides
+        whether or not to add nodes to the cluster. Returns the number of nodes
+        to add.
+        """
+        need_to_add = self._eval_required_slots_hour()
```

Comment:
where is this function?
Maybe I'm reading this wrong, but it looks as if `_eval_required_slots_hour`  should be `_eval_required_instances` or the opposite.